### ISSUE TITLE
closure compiler can introduce non-ES5 constructs

### ISSUE THREAD

**Issue by sbc100** [emscripten-core/emscripten#12628]
We suspected that this may be the case when we switched from using `--language_out NO_TRANSPILE` and `--language_in ECMASCRIPT_2020` but I've found an actual example now:

```
$ cat test.js 
mergeInto(LibraryManager.library, {
  foo: function(x, y) {
    var f = {
     'foo': foo,
     'bar': bar
    };
    return f;
  }
});
$ cat test.c 
extern void foo();
#include <stdio.h>

int main() {
  foo();
  printf("done\n");
  return 0;
}
$ ./emcc ./test.c --js-library=test.js --closure=1
$ ./node_modules/.bin/es-check es5 a.out.js
          ES-Check Error:
          ----
          · erroring file: a.out.js
          · error: SyntaxError: Unexpected token (929:13)
          · see the printed err.stack below for context
          ----
```

The problem is that the output file uses the shorthand syntax for object literal:

```
function _foo(a, b) {                                                               
  return {foo, bar};                                                                                                       
}   
```

**Comment by sbc100** [comment]
My proposal for how to fix is to address it by fixing #11984.   

The idea is that we have conservative default value for INPUT_JS_VERSION and OUTPUT_JS_VERSION that users can control.

When they differ, we transpile using closure-compiler automatically.

Sadly because we want to be conservative by default any users who want to use non-ES5 constrcuts in their input (ES_ASM, EM_ASM, --library-js) will need to opt into the more recent version with `-s INPUT_JS_VERSION`.

**Comment by sbc100** [comment]
This is no longer an issue as we output non-ES5 by default now anyway.
